crates/pathfinder/src/cairo/ext_py.rs

```diff
@@ -520,6 +537,110 @@ mod tests {
         jh.await.unwrap();
     }
 
+    #[test_log::test(tokio::test)]
+    #[ignore = "needs python venv"]
+    async fn call_with_pending_updates() {
+        use crate::sequencer::reply::StateUpdate;
+
+        let db_file = tempfile::NamedTempFile::new().unwrap();
+
+        let s = crate::storage::Storage::migrate(
+            PathBuf::from(db_file.path()),
+            crate::storage::JournalMode::WAL,
+        )
+        .unwrap();
+
+        let mut conn = s.connection().unwrap();
+        conn.execute("PRAGMA foreign_keys = off", []).unwrap();
+
+        let tx = conn.transaction().unwrap();
+
+        fill_example_state(&tx);
+
+        tx.commit().unwrap();
+
+        let (shutdown_tx, shutdown_rx) = oneshot::channel::<()>();
+
+        let (handle, jh) = super::start(
+            PathBuf::from(db_file.path()),
+            std::num::NonZeroUsize::new(1).unwrap(),
+            async move {
+                let _ = shutdown_rx.await;
+            },
+            // chain doesn't matter here because we are not estimating any real transaction
```

Comment:
```suggestion
```

Don't look, didn't copypaste. I was hoping to merge #422 here but perhaps in future someone else will adapt that, or new ideas come forth.